### PR TITLE
test: Mark flaky tests with `run_alone` to reduce CI failures

### PR DESCRIPTION
**Pull request by vdusek**
Mark the following tests with `@pytest.mark.run_alone` so they run in isolation (numprocesses=1) rather than in parallel, reducing flakiness observed across the last ~10 master commits.

### CODE REVIEW

File: tests/unit/_autoscaling/test_snapshotter.py [hunk 1]
```diff
@@ -212,6 +213,9 @@ async def test_methods_raise_error_when_not_active() -> None:
         assert snapshotter.active is True
 
 
+@pytest.mark.skip(
+    reason='Flaky due to snapshot pruning boundary condition, see https://github.com/apify/crawlee-python/issues/1734'
+)
 async def test_snapshot_pruning_removes_outdated_records(
```
Comment (Pijukatel) [inline tests/unit/_autoscaling/test_snapshotter.py:219]: Have you seen any flakiness after this was merged?
https://github.com/apify/crawlee-python/pull/1716

Comment (Pijukatel) [reply tests/unit/_autoscaling/test_snapshotter.py:219]: I think it should be fixed by https://github.com/apify/crawlee-python/pull/1716/changes#diff-87bab5034cd4fe2a86d4fbbfd10dc2d989df12e817396f3837e99cd7a73464f1R34

Comment (Pijukatel) [reply tests/unit/_autoscaling/test_snapshotter.py:219]: Ok, I see this branch was properly based on that change

Comment (vdusek) [reply tests/unit/_autoscaling/test_snapshotter.py:219]: yes, in this PR
